static/app/views/alerts/rules/metric/details/metricChartOption.tsx

```diff
@@ -164,12 +165,20 @@ export function getMetricAlertChartOption({
   handleIncidentClick,
   showWaitingForData,
 }: MetricChartData): MetricChartOption {
-  const criticalTrigger = rule.triggers.find(
-    ({label}) => label === AlertRuleTriggerType.CRITICAL
-  );
-  const warningTrigger = rule.triggers.find(
-    ({label}) => label === AlertRuleTriggerType.WARNING
-  );
+  let criticalTrigger: Trigger | undefined;
+  let warningTrigger: Trigger | undefined;
+
+  for (const trigger of rule.triggers) {
+    if (!criticalTrigger && trigger.label === AlertRuleTriggerType.CRITICAL) {
+      criticalTrigger = trigger;
+    }
```

Comment:
Nit: Might be a little bit more readable. I'm not sure.

```suggestion
    if (trigger.label === AlertRuleTriggerType.CRITICAL) {
      criticalTrigger ??= trigger;
    }
```